source/fundamentals/data-formats/document-data-format-data-class.txt

```diff
@@ -15,4 +15,155 @@ Document Data Format: Data Classes
 Overview
 --------
 
-TODO: add content. right now making a container page for the ref used in the quickstart
+In this guide, you can learn how to store and retrieve data in the
+{+driver-long+} using **Kotlin data classes**.
+
+Serialize and Deserialize a Data Class
+--------------------------------------
+
+The driver natively supports encoding and decoding Kotlin data classes for
+MongoDB read and write operations using the **default codec registry**. The
+default codec registry is a collection of classes called **codecs** that
+define how to convert encode and decode Kotlin types.
+
+.. _fundamentals-example-data-class:
+
+Example Data Class
+~~~~~~~~~~~~~~~~~~
+
+The code examples in this section reference the following sample data class, which
+describes a data storage device:
+
+.. literalinclude:: /examples/generated/DataClassTest.snippet.data-class.kt
+   :language: kotlin
+
+Insert a Data Class
+~~~~~~~~~~~~~~~~~~~
+
+You can insert a ``DataStorage`` instance as shown in the following code:
+
+.. literalinclude:: /examples/generated/DataClassTest.snippet.insert-data-class.kt
+   :language: kotlin
+
+Retrieve a Data Class
+~~~~~~~~~~~~~~~~~~~~~
+
+You can retrieve documents as ``DataStorage`` instances and print them
+as shown in the following code:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /examples/generated/DataClassTest.snippet.retrieve-data-class.kt
+      :language: kotlin
+
+   .. output::
+      :language: console
+
+      DataStorage(productName=tape, capacity=5.0)
+
+.. _fundamentals-data-class-annotations:
+
+Specify Component Conversion Using Annotations
+----------------------------------------------
+
+This section describes the annotations you can use to configure the
+serialization behavior of data classes and provides an example to
+demonstrate the annotation behavior.
+
+You can use the following annotations on data classes:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 10
+
+   * - Annotation Name
+     - Description
+
+   * - ``BsonId``
+     - Marks a property to serialize as the _id property.
+
+   * - ``BsonProperty``
+     - Specifies a custom document field name when converting the data class
+       field to BSON. You can include a discriminator to serialize data classes
```

Comment:
Remove the sentence about discriminators.  The data class codec (and the record codec, for that matter), ignore the `useDiscriminator` field of `BsonProperty`.